source/model/backtraceMetrics.ts

```diff
@@ -0,0 +1,228 @@
+import { BacktraceClientOptions } from '..';
+import { currentTimestamp, getBacktraceGUID, getEndpointParams, uuid } from '../utils';
+import { APP_NAME, USER_AGENT, VERSION } from '../consts/application';
+import { SEC_TO_MILLIS } from '../consts';
+
+/**
+ * Handles Backtrace Metrics.
+ */
+export class BacktraceMetrics {
+  private readonly universe: string;
+  private readonly token: string;
+  private readonly timeout: number  = 15000; // Fifteen seconds in milliseconds.
+  private readonly persistenceInterval: number  = 1800000; // Thirty minutes in milliseconds.
+  private readonly heartbeatInterval: number  = 60000; // One minutes in milliseconds.
+
+  private readonly timestamp = currentTimestamp()
+  private readonly userAgent = USER_AGENT;
+  private readonly applicationName = APP_NAME;
+  private readonly applicationVersion = VERSION;
+
+  private readonly guid = getBacktraceGUID();
+
+  public hostname = 'https://events.backtrace.io';
+
+  constructor(configuration: BacktraceClientOptions) {
+    if (!configuration.endpoint) {
+      throw new Error(`Backtrace: missing 'endpoint' option.`);
+    }
+    const {universe, token} = getEndpointParams(configuration.endpoint)
+
+    if(!universe) {
```

Comment:
did we apply formatting rules? The if statement should have an additional space before the bracket `(`